source/connect.txt

```diff
@@ -123,3 +141,88 @@ For details, see :ref:`comp-command-line-connect`.
+
+Connection String Examples
+--------------------------
+
+The following examples highlight some advanced connection string options.
+For information on all connection string options, see :ref:`connections-connection-options`.  
+
+TLS Options
+~~~~~~~~~~~
+
+The following connection string enables TLS using the ``tls`` URI option. 
+``tlsCertificateKeyFile`` specifies the path to the client certificate and 
+private key. Specifying the ``tlsCertificateKeyFile`` in your connection string
+is optional, unless the server requires client certificates, or if 
+you are using X.509 authentication.
+
+.. code-block:: 
+
+   mongodb://myDatabaseUser:D1fficultP%40ssw0rd@db0.example.com/?directConnection=true&tls=true&tlsCAFile=path/to/rootCA.crt&tlsCertificateKeyFile=path/to/server_certificate.pem
+
+To enable SSL, use the equivalent ``ssl=true`` option in place of ``tls=true``. 
+
+X.509 Authentication
+````````````````````
+
+The following example demonstrates how to specify X.509 authentication in your connection string
+using the :urioption:`authMechanism` connection string option:
+
+.. code-block:: 
+
+   mongodb://myDatabaseUser:D1fficultP%40ssw0rd@db1.example.com/?authMechanism=MONGODB-X509&tls=true&tlsCertificateKeyFile=%2Fpath%2Fto%2Fclient_certificate.pem
+
+LDAP
+~~~~
+
+The following example specifies LDAP authentication. It sets :urioption:`authMechanism` to ``plain`` 
+to specify the use of the LDAP authentication mechanism, and sets ``authSource`` to ``$external``
+to indicate that the authentication database is ``$external``.
+
+.. code-block:: 
+
+   mongodb://myDatabaseUser:D1fficultP%40ssw0rd@db2.example.com/?authMechanism=PLAIN&authSource=$external
+
+Kerberos
+~~~~~~~~
+
+The following example specifies Kerberos authentication. It sets :urioption:`authMechanism` to
+``GSSAPI`` to use the Kerberos authentication mechanism, and sets ``authSource`` to ``$external``
+to indicate that the authentication database is ``$external``.
+
+.. code-block:: 
+
+   mongodb://myDatabaseUser@db3.example.com/?authMechanism=GSSAPI&authSource=$external
+
+Read/Write Preferences
+~~~~~~~~~~~~~~~~~~~~~~
+
+To enable read/write preferences in your connection string, you can use the 
+``readPreference`` option for read preferences and the ``w`` option for write concerns.
+
+The following connection string sets both read and write preferences. 
+``readPreference=secondary`` specifies that read operations should be directed to secondary 
+members of the replica set, and ``w=majority`` ensures that write operations are 
+acknowledged by the majority of the replica set members.
+
+.. code-block::
+
+   mongodb://myDatabaseUser:D1fficultP%40ssw0rd@db4.example.com/database?readPreference=secondary&w=majority
+
+Analytic Nodes
+~~~~~~~~~~~~~~
+
+To specify an ``analytics`` node type in your connection string, set your ``readPreference`` to
+``secondary`` to specify that the read operations should be directed to secondary nodes, then
```

Comment:
[nit]

Similar to above, consider removing "," on line 217 and replacing "then" with "and".

Reply:
If this one seems to long as a single statement, consider bulleting it out.